src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs

```diff
@@ -427,23 +427,23 @@ public async Task MapAction_NoJsonTypeInfoResolver_ThrowsException()
         Assert.Equal("JsonSerializerOptions instance must specify a TypeInfoResolver setting before being marked as read-only.", exception.Message);
     }
 
-    public static IEnumerable<object[]> NullResult
+    public static IEnumerable<object[]> NullResultWithNullAnnotation
     {
         get
         {
             return new List<object[]>
             {
                 new object[] { "IResult? () => null", "The IResult returned by the Delegate must not be null." },
                 new object[] { "Task<IResult?>? () => null", "The Task returned by the Delegate must not be null." },
-                new object[] { "Task<bool?>? () => null", "The Task returned by the Delegate must not be null." },
+                new object[] { "Task<bool>? () => null", "The Task returned by the Delegate must not be null." },
```

Comment:
Can we retain the existing `Task<bool?>?` scenario here? I don't believe the change you made will break this and we want the right thing to happen regardless.

Reply:
We sure can